web/client/plugins/widgetbuilder/ChartLayerSelector.jsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { branch, compose } from 'recompose';
+import { Glyphicon } from "react-bootstrap";
+import { createSelector } from 'reselect';
+import castArray from "lodash/castArray";
+import isEmpty from "lodash/isEmpty";
+
+import { HTML, Message } from '../../components/I18N/I18N';
+import BorderLayout from '../../components/layout/BorderLayout';
+import Toolbar from '../../components/widgets/builder/wizard/common/layerselector/Toolbar';
+import InfoPopover from '../../components/widgets/widget/InfoPopover';
+import ButtonRB from "../../components/misc/Button";
+import tooltip from "../../components/misc/enhancers/tooltip";
+import { selectedCatalogSelector } from '../../selectors/catalog';
+import BuilderHeader from './BuilderHeader';
+import CatalogComp from './Catalog';
+
+const Button = tooltip(ButtonRB);
+
+const Catalog = compose(
+    branch(
+        ({catalog} = {}) => !catalog,
+        connect(createSelector(selectedCatalogSelector, catalog => ({catalog})))
+    )
+)(CatalogComp);
```

Comment:
I'm seeing the use of recompose a lot in this PR,
do we really need it also in new components like this one?
please replace it with a react api only solution if possible